### DISCUSSION TITLE
Is React 18 a prerequisite?

### DISCUSSION THREAD

**Discussion by riha** [resend/react-email#1311] [Q&A] [answered]
I guess I should know this 😬 But is there a way of getting this to run on NextJS 12 and react 17? Or is React 18 a requirement?

**Comment by gabrielmfern** [comment]
I wish it wasn't, but at least for TypeScript, it is. This is because we use a specific version of React's types ourselves
and that, a lot of times, causes issues with React saying this or that isn't a valid node or element.

**Reply by gabrielmfern** [reply]
Not exactly sure how we would get that to work properly regardless of the types's version though, even Next has these kinds of issues.

**Comment by gabrielmfern** [comment, marked as answer]
What I would recommend you do is have a different `package.json` for your emails, maybe with the `emails` directory inside of a *transactional/emails* and the `package.json` inside of `transactional`, and then there use React 18 with its types.
This should work fine, at least until you render it inside of the Next app, as the rendering process for React Email is not one with React's.

**Reply by bentice** [reply]
So we can make the emails but we couldn't use them in our NextJS app because they wouldn't render unless we are on React 18? I think this basically means React 18 is a prerequisite.

I'm also currently trying to use react-email on a non react 18 project.

**Reply by gabrielmfern** [reply]
What do you mean that they wouldn't render? `@react-email/render` will bring React 18 itself IIRC so it should work, only issue is the types for elements that might mismatch.
